Assignment3/Assignment3/ViewController.swift

```diff
@@ -11,57 +11,161 @@ import UIKit
 class ViewController: UIViewController {
 
     //    Задача 1. Создать строку с своим именем, вывести количество символов содержащихся в ней.
-
+    
     func countStringCharacters(name: String) -> Int {
-        fatalError()
+        return name.count
     }
-
+    
     //    Задача 2. Создать строку с своим отчеством проверить его на окончание “ич/на”
-
+    
     func checkPatronomic(name: String) -> Bool {
-        fatalError()
+        if name.hasSuffix("ich") {
+            return true
+        }
+        if name.hasSuffix("vna") {
+            return true
+        }
+        else {
+            return false
+        }
     }
-
+    
     //    Задача 3. Cоздать строку, где слитно написано Ваши ИмяФамилия “IvanVasilevich"
-
+    
     func divideCredentials(fullName: String) -> String {
-        fatalError()
+      var newStringArray: [String] = []
+      for character in fullName {
+          if String(character) == String(character).uppercased() {
+              newStringArray.append(" ")
+          }
+          newStringArray.append(String(character))
+      }
+      let finalResult = newStringArray.joined(separator: "").substring(from: 1)
+      return finalResult
     }
-
+    
     //    Задача 4. Вывести строку зеркально Ось → ьсО не используя reverse
-
+    
     func mirrorString(input: String) -> String {
-        fatalError()
+        var str = ""
+        for character in input {
+            str = "\(character)" + str
+        }
+        return str
     }
-
+    
     //    Задача 5. Добавить запятые в строку как их расставляет калькулятор
-
+    
     func formatNumber(input: String, separator: String) -> String {
-        fatalError()
+        var newText : String = ""
+        for character in input {
+            newText.insert(character, at: newText.startIndex)
+        }
+        var result : String = ""
+        for (index , number) in newText.enumerated() {
+            if index != 0 && index % 3 == 0 {
+                result.append(separator)
+            }
+            result.append(number)
+        }
+        var finalResult : String = ""
+        for i in result {
+            finalResult.insert(i, at: finalResult.startIndex)
+        }
+        return finalResult
     }
-
+    
     //    Задача 6. Проверить пароль на надежность от 1 до 5
-
+    
     func checkPassword(password: String) -> Int {
-        fatalError()
+        var markNumbers = 0
+        var markLetter = 0
+        var markUppercased = 0
+        var markSymbol = 0
+        var totalScore = 0
+        for character in password {
+            if character.isNumber {
+                markNumbers += 1
+            }
+            if character.isLetter {
+                markLetter += 1
+            }
+            if character.isUppercase {
+                markUppercased += 1
+            }
+            if character.isPunctuation {
+                markSymbol += 2
```

Comment:
White +2 not +1?

Reply:
As in the last password, there were only 4 possible variants to make +1, and in the task, you should get +5 in total, I understood it like +2 for Symbols